source/includes/fundamentals/code-snippets/BulkWrite.java

```diff
@@ -0,0 +1,224 @@
+
+    private void bulkWriteNotOrderedExample() {
+        List<WriteModel<Document>> bulkOperations = new ArrayList<>();
+
+        InsertOneModel<Document> doc1 = new InsertOneModel<>(new Document("_id", 3));
+        ReplaceOneModel<Document> doc2 = new ReplaceOneModel<>(Filters.eq("_id", 1), 
+                                            new Document("_id", 1).append("x", 2));                            
+        UpdateOneModel<Document> doc3 = new UpdateOneModel<>(Filters.eq("_id", 3), Updates.set("x", 2));
+        DeleteManyModel<Document> doc4 = new DeleteManyModel<>(Filters.eq("x", 2));
+   
+        bulkOperations.add(doc1);
+        bulkOperations.add(doc2);
+        bulkOperations.add(doc3);
+        bulkOperations.add(doc4);
+        
+        // begin bulkWriteNotOrderedExample
+        BulkWriteOptions options = new BulkWriteOptions().ordered(false);
+        
+        collection.bulkWrite(bulkOperations, options); 
+        //end bulkWriteNotOrderedExample
+    }
+
+    private void bulkWriteExample() {
+        // begin bulkWriteExample
+
+        List<WriteModel<Document>> bulkOperations = new ArrayList<>();
+
+        InsertOneModel<Document> doc1 = new InsertOneModel<>(new Document("_id", 3));
+        ReplaceOneModel<Document> doc2 = new ReplaceOneModel<>(Filters.eq("_id", 1), 
+                                            new Document("_id", 1).append("x", 2));                            
+        UpdateOneModel<Document> doc3 = new UpdateOneModel<>(Filters.eq("_id", 3), Updates.set("x", 2));
+        DeleteManyModel<Document> doc4 = new DeleteManyModel<>(Filters.eq("x", 2));
+   
+        bulkOperations.add(doc1);
+        bulkOperations.add(doc2);
+        bulkOperations.add(doc3);
+        bulkOperations.add(doc4);
+
+        collection.bulkWrite(bulkOperations);               
+        //end bulkWriteExample
+    }
+
+    private void insertDocumentsExample(){
+        collection.drop();
+        List<WriteModel<Document>> bulkOperations = new ArrayList<>();
+
+        // begin insertDocumentsExample
+        InsertOneModel<Document> doc1 = new InsertOneModel<>(new Document("_id", 3));
+        InsertOneModel<Document> doc2 = new InsertOneModel<>(new Document("_id", 4));
+        //end insertDocumentsExample
+        
+        bulkOperations.add(doc1);
+        bulkOperations.add(doc2);
+
+        collection.bulkWrite(bulkOperations);
+    }
+    
+    private void replaceDocumentsExample(){
+        collection.drop();
+        List<WriteModel<Document>> bulkOperations = new ArrayList<>();
+
+        InsertOneModel<Document> doc1 = new InsertOneModel<>(new Document("_id", 1));
+        InsertOneModel<Document> doc2 = new InsertOneModel<>(new Document("_id", 2));
+
+        // begin replaceDocumentsExample
+        ReplaceOneModel<Document> doc3 = new ReplaceOneModel<>(
+                                            Filters.eq("_id", 1), 
+                                            new Document("_id", 1).append("x", 4) );
```

Comment:
Nit: Very minor code style nit - there is extra whitespace between the brackets.